src/lib.rs

```diff
@@ -0,0 +1,192 @@
+use regex::Regex;
+use std::io::Result;
+use std::process::exit;
+use std::process::{Child, Command};
+
+use crate::options::Opt;
+
+#[macro_use]
+extern crate log;
+
+mod git;
+pub mod options;
+
+/// Enumeration to store exit codes. The first one, Success is by default set to 0
+enum ExitCode {
+    Success,
+    NotInAGitRepository,
+    NoRemoteMatching,
+    NoRemoteAvailable,
+    NotAbleToOpenSystemBrowser,
+    BrowserNotAvailable,
+}
+
+const DEFAULT_REMOTE_ORIGIN: &str = "origin";
+const BITBUCKET_HOSTNAME: &str = "bitbucket.org";
```

Comment:
Maybe you should create an `enum` for extracting git provider specific behaviors.

```rust
enum GitProvider{
    Github,
   Bitbucket
}
```